optuna/integration/botorch.py

```diff
@@ -119,11 +120,24 @@ def logei_candidates_func(
 
     if train_obj.size(-1) != 1:
         raise ValueError("Objective may only contain single values with logEI.")
+
     if train_con is not None:
-        raise ValueError(
-            "Constraint is not supported with logei_candidates_func. "
-            + "Please use qei_candidates_func instead."
-        )
+        train_y = torch.cat([train_obj, train_con], dim=-1)
+
+        is_feas = (train_con <= 0).all(dim=-1)
+        train_obj_feas = train_obj[is_feas]
+
+        if train_obj_feas.numel() == 0:
+            # TODO(hvy): Do not use 0 as the best observation.
+            _logger.warning(
+                "No objective values are feasible. Using 0 as the best objective in qEI."
+            )
+            best_f = torch.zeros(())
+        else:
+            best_f = train_obj_feas.max()
+
+        n_constraints = train_con.size(1)
```

Comment:
This variable should be defined near the use of it. How about moving this to right before the line 152?